Entities/SceneLayer.cpp

```diff
@@ -505,41 +505,42 @@ namespace RTE {
 
 	template <>
 	void SceneLayerImpl<true>::RegisterDrawing(int left, int top, int right, int bottom) {
-		m_Drawings.push_back(IntRect(left, top, right, bottom)); 
+		m_Drawings.emplace_back(left, top, right, bottom); 
```

Comment:
No, why are you doing this? This is unnecessary, and IMO less clear. Move semantics and std::forward make this no more efficient than a push_back().

Reply:
Linter - https://rules.sonarsource.com/cpp/RSPEC-6003

Reply:
[Well that linter is wrong](https://github.com/microsoft/STL/blob/f8b8ad6dea901d360d8165cc6719de47ca9d9ca2/stl/inc/vector#L874)